src/pages/index.js

```diff
@@ -105,88 +193,98 @@ function openProfileEditModal() {
+      userInfo.setUserInfo(info);
+      editProfilePopup.close();
+    })
+    .catch((error) => {
+      console.log(error);
+    })
+    .finally(() => {
+      editProfilePopup.renderLoading(false);
+    });
 }
 
-profileEditButton.addEventListener("click", openProfileEditModal);
-
-// Add New Card Modal
+// Add New Card Modal /////////////////////////////////////////////
 
 addNewCardButton.addEventListener("click", () => newCardPopup.open());
-cardList.renderItems();
-
-function handleAddNewCardSubmit(inputs) {
-  const cardData = {
-    name: inputs.title,
-    link: inputs.description,
-  };
-
-  createCard(cardData);
-  newCardPopup.close();
-  addFormValidator.disableButton();
-  // thanks
-}
 
-function generateCard(cardData) {
-  const card = new Card(cardData, "#card-template", handlePreviewModal);
-  return card.generateCard();
+function handleAddNewCardSubmit(cardData) {
+  newCardPopup.renderLoading(true);
+  api
+    .addNewCard(cardData)
+    .then((card) => {
+      createCard(card);
+      newCardPopup.close();
+      addFormValidator.disableButton();
+    })
+    .catch((error) => {
+      console.log(error);
+    })
+    .finally(() => {
+      newCardPopup.renderLoading(false, "Save");
+    });
 }
 
 function createCard(cardData) {
   const cardElement = generateCard(cardData);
   cardList.addItem(cardElement);
 }
 
-// function renderCard(cardElement, method = "prepend") {
-//   cardsList[method](cardElement);
-// }
-
 function handlePreviewModal(data) {
   imagePreviewPopup.open({
     link: data.link,
     name: data.name,
   });
 }
 
-const editFormValidator = new FormValidator(config, profileForm);
-editFormValidator.enableValidation();
-
-const addFormValidator = new FormValidator(config, addNewCardFrom);
-addFormValidator.enableValidation();
-
-// // Form Valiation
-
-// // define an object for storing validators
-// const formValidators = {};
-
-// const enableValidation = (config) => {
-//   const formList = Array.from(document.querySelectorAll(config.formSelector));
-//   formList.forEach((formElement) => {
-//     const validator = new FormValidator(config, formElement);
-//     // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
-//     const profileForm = formElement.getAttribute("profile-form");
-//     const addNewCardFrom = formElement.getAttribute("add-card-form");
-
-//     // Here you store the validator using the `name` of the form
-//     formValidators[profileForm] = validator;
-//     formValidators[addNewCardFrom] = validator;
-
-//     validator.enableValidation();
-//   });
-// };
+function handleDeleteClick(card) {
+  confirmDeleteModal.onSubmit(() => {
+    deleteCard(card);
+  });
+  confirmDeleteModal.open();
+}
+addNewCardButton.addEventListener("click", () => newCardPopup.open());
 
-// enableValidation(config);
+function deleteCard(card) {
+  api.deleteCard(card._id).then((result) => {
+    card.handleDeleteCard();
+  });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**